### PR TITLE
Support BSD 'X-Clause BSD' license string

### PR DESCRIPTION
**Pull request by awesomebytes**
As mentioned in https://github.com/ros-infrastructure/superflore/issues/163

### CODE REVIEW

File: superflore/utils.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def trim_string(string, length=80):
 
 
 def get_license(l):
-    bsd_re = '^(BSD)((.)*([124]))?'
+    bsd_re = '^(BSD)((.)*([124]))?|([1234])(.)?Clause(.)*(BSD)'
```
Comment (awesomebytes) [inline superflore/utils.py]: @allenh1 My regex skills are pretty dusty, I tried to make it as safe as possible... so I just added a second option to match, the original one is untouched.

File: superflore/utils.py [hunk 2]
```diff
@@ -170,9 +170,12 @@ def get_license(l):
             return 'Apache-%.1f' % (float(version))
         return 'Apache-1.0'
     elif re.search(bsd_re, l, f):
-        version = re.search(bsd_re, l, f).group(4)
+        matches = re.search(bsd_re, l, f)
+        version = matches.group(4)
+        if not version:
+            version = matches.group(5)
```
Comment (awesomebytes) [inline superflore/utils.py]: @allenh1  And here I just try to get the version from either the previous kind of match, or the new (giving priority to the previous case).

**Comment by allenh1** [comment]
```
superflore/utils.py:178:45: W291 trailing whitespace\n1\n'
```

The linter is angry.

**Comment by allenh1** [comment]
Other than the linter failures, I would be really grateful if you could add some tests for the license check. You should be able to find some here.

https://github.com/ros-infrastructure/superflore/blob/adf2fc5d9cccc2fdb92a653718e96b74534b6b74/tests/test_utils.py#L82-L123

This just helps prevent this regex getting broken in the future.

**Comment by awesomebytes** [comment]
> ```
> superflore/utils.py:178:45: W291 trailing whitespace\n1\n'
> ```
> The linter is angry.

Fixed

**Comment by allenh1** [comment]
@awesomebytes did you mean to close this?

**Comment by awesomebytes** [comment]
I messed up the PR. And I messed up my Python installation so I can't run superflore (neither the tests) anymore.

If you wanna give it a try to fix it... you are welcome, otherwise I'll need to revisit this some other day.

Note that I just noticed a logic error. If `'3-Clause BSD'` is the license string, by your previous code you return `BSD` but I was returning `BSD-3`. So there must be another `if matches.group(5) == 3: return 'BSD'` or something like that.

Sorry for the mess. I should go to sleep.

**Comment by allenh1** [comment]
@awesomebytes no worries -- we can just fix it at some other point.

**Pull request closed without merging**